src/main/java/examples/CassandraClientExamples.java

```diff
@@ -0,0 +1,177 @@
+import io.vertx.cassandra.ResultSet;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+
+import java.util.List;
+
+public class CassandraClientExamples {
+
+  public void specifyingNodes(Vertx vertx) {
+    CassandraClientOptions options = new CassandraClientOptions();
+    options
+      .addContactPoint("node1.address")
+      .addContactPoint("node2.address")
+      .addContactPoint("node3.address");
+    CassandraClient client = CassandraClient.create(vertx, options);
+  }
+
+  public void connecting(CassandraClient cassandraClient) {
+    cassandraClient.connect(connect -> {
+      if (connect.succeeded()) {
+        System.out.println("Just connected");
+      } else {
+        System.out.println("Unable to connect");
+        connect.cause().printStackTrace();
+      }
+    });
+  }
+
+  public void disconnecting(CassandraClient cassandraClient) {
+    cassandraClient.disconnect(disconnect -> {
+      if (disconnect.succeeded()) {
+        System.out.println("Just disconnected");
+      } else {
+        System.out.println("Unable to disconnect");
+        disconnect.cause().printStackTrace();
+      }
+    });
+  }
+
+  public void lowLevelQuerying(CassandraClient cassandraClient) {
+    cassandraClient.execute("SELECT * FROM my_keyspace.my_table where my_key = 'my_value'", execute -> {
+      if (execute.succeeded()) {
+        ResultSet resultSet = execute.result();
+
+        resultSet.one(one -> {
+          if (one.succeeded()) {
+            Row row = one.result();
+            System.out.println("One row successfully fetched");
+          } else {
+            System.out.println("Unable to fetch a row");
+            one.cause().printStackTrace();
+          }
+        });
+
+        resultSet.fetchMoreResults(fetchMoreResults -> {
+          if (fetchMoreResults.succeeded()) {
+            int availableWithoutFetching = resultSet.getAvailableWithoutFetching();
+            System.out.println("Now we have " + availableWithoutFetching + " rows fetched, but not consumed!");
+            if (resultSet.isFullyFetched()) {
+              System.out.println("The result is fully fetched, we don't need to call this method for one more time!");
+            } else {
+              System.out.println("The result still does not fully fetched");
+            }
+          } else {
+            System.out.println("Unable to fetch more results");
+            fetchMoreResults.cause().printStackTrace();
+          }
+        });
+
+      } else {
+        System.out.println("Unable to execute the query");
+        execute.cause().printStackTrace();
+      }
+    });
+  }
+
+  public void streaming(CassandraClient cassandraClient) {
+    cassandraClient.queryStream("SELECT * FROM my_keyspace.my_table where my_key = 'my_value'", queryStream -> {
+      if (queryStream.succeeded()) {
+        CassandraRowStream stream = queryStream.result();
+        stream.handler(row -> {
+          // handler will be called with row
```

Comment:
Can you add a statement to show how to extract data from the row and push it to an http server response?

You would also use the response object `isWwriteQueueFull` and `drainHandler` methods to show `pause` and `resume` use in context.

Reply:
Nice suggestion. Just done.